synfig-core/src/synfig/node.cpp

```diff
@@ -202,8 +204,11 @@ Node::get_guid()const
 	std::lock_guard<std::mutex> lock(guid_mutex_);
 	if(!guid_)
 	{
-		guid_.make_unique();
-		global_node_map().add(guid_, const_cast<Node*>(this));
+		bool added = false;
```

Comment:
Maybe shorten this code?
```
do {
	guid_.make_unique();
} while (!global_node_map().add(guid_, const_cast<Node*>(this)));
```

Reply:
Yeah, I thought it this way, but maybe not that readable.
Compiler would throw that variable away anyway, right?

Reply:
> Compiler would throw that variable away anyway, right?
Yes.